test/WebJobs.Script.Tests/Binding/ActionResults/RawScriptResultTests.cs

```diff
@@ -60,10 +60,7 @@ public async Task AddsHttpCookies()
                 Headers = new Dictionary<string, object>(),
                 Cookies = new List<Tuple<string, string, CookieOptions>>()
                 {
-                    new Tuple<string, string, CookieOptions>("firstCookie", "cookieValue", new CookieOptions()
-                    {
-                        SameSite = SameSiteMode.None
```

Comment:
I believe in the v3 branch, @mhoeger cahnged this to be `SameSideMode.Lax`, which the test then passed with. Does this pass?

Reply:
Yes. This is the right approach for this branch. 

There was a loooong discussion on this :)

Reply:
I think this might be a little off - the way it's written in [RpcHttpConverter](https://github.com/Azure/azure-functions-host/blob/645e39bc83a4f92c13bb51130e0d26bba007f0a7/src/WebJobs.Script/Workers/Rpc/MessageExtensions/RpcMessageExtensionUtilities.cs#L42), `SameSite = SameSiteMode.None` will always be present as a default value, and SameSite as unspecified will never exist. 

So I think "secondCookie" is an invalid test and should have SameSite set to some value, which saves us from the bug we found in the Append extension method that adds "samesite=lax" when unspecified.

If I'm remembering right, the behavior of SameSite = SameSiteMode.None became an explicit addition of `sameSite=none`. I think we might need to apply a [similar fix to V3](https://github.com/Azure/azure-functions-host/pull/5331/files), where we change the default value of cookies that come from an RPC message to SameSiteMode.Unspecified, and update the test so that we don't test `SameSite = null`. Does that sound right?

Reply:
Your understanding of the behavior is correct, but this test wasn't validating that properly. `SameSite.None` will add `samesite=none` to the cookie.

I think we're missing the proper tests, but without the middleware registered (which is the case with the runtime) and with the pattern/overload we use to add the cookies returned by the workers, we're maintaining the right behavior. 

Let me take a closer look and make sure the tests reflect what we want to validate.

Reply:
Also, the test change in the other PR just validate that explicitly setting that here would be honored, which will be the case with the current state.

Reply:
So to answer Brett's initial question... if we change this to `lax`, that will be correctly set.

@mhoeger we need a proper test that validates the default from the `RpcHttpConverter` (I'm not sure whether we have that), as updating this to use unspecified just relies on that behavior (but doesn't validate it)

Reply:
I've updated the test to mirror what we were validating in 3.0. Also moved the conversion changes, but as mentioned above, we need a proper test for that, as this test doesn't validate that behavior.

Reply:
Definitely agreed! Adding this issue: https://github.com/Azure/azure-functions-host/issues/5384 and linking it to this one that's planned for this sprint: https://github.com/Azure/azure-functions-host/issues/4890